gridscale/resource_gridscale_template_test.go

```diff
@@ -0,0 +1,136 @@
+package gridscale
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	"github.com/gridscale/gsclient-go"
+)
+
+func TestAccResourceGridscaleTemplate_Basic(t *testing.T) {
+	var object gsclient.Template
+	name := fmt.Sprintf("object-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGridscaleTemplateDestroyCheck,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckResourceGridscaleTemplateConfig_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourceGridscaleTemplateExists("gridscale_template.foo", &object),
+					resource.TestCheckResourceAttr(
+						"gridscale_template.foo", "name", name),
+				),
+			},
+			{
+				Config: testAccCheckResourceGridscaleTemplateConfig_basic_update(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourceGridscaleTemplateExists("gridscale_template.foo", &object),
+					resource.TestCheckResourceAttr(
+						"gridscale_template.foo", "name", "newname"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckResourceGridscaleTemplateExists(n string, object *gsclient.Template) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No object UUID is set")
+		}
+
+		client := testAccProvider.Meta().(*gsclient.Client)
+
+		id := rs.Primary.ID
+
+		foundObject, err := client.GetTemplate(emptyCtx, id)
+
+		if err != nil {
+			return err
+		}
+
+		if foundObject.Properties.ObjectUUID != id {
+			return fmt.Errorf("Object not found")
+		}
+
+		*object = foundObject
+
+		return nil
+	}
+}
+
+func testAccCheckGridscaleTemplateDestroyCheck(s *terraform.State) error {
+	client := testAccProvider.Meta().(*gsclient.Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "gridscale_ipv4" {
```

Comment:
I think this should be `gridscale_template`  resource.

Reply:
I've already fixed it in another branch (will open a PR after this PR)

Reply:
:+1: